test/common/pixeldiff.html

```diff
@@ -571,33 +573,41 @@
 
     const img_now = document.createElement("img");
     img_now.setAttribute("src", base + "-pixels.png");
-    img_now.setAttribute("width", "100%");
 
     const img_ref = document.createElement("img");
     img_ref.setAttribute("src", base + "-reference.png");
-    img_ref.setAttribute("width", "100%");
 
     const img_delta = document.createElement("img");
     img_delta.setAttribute("src", base + "-delta.png");
-    img_delta.setAttribute("width", "100%");
-
-    img_now.addEventListener("load", () => {
-        const diff = document.getElementById("diff");
-        diff.style.width = img_now.naturalWidth + "px";
-        diff.style.height = img_now.naturalHeight + "px";
-        const ecompare = document.createElement("div");
-        ecompare.classList.add("pixelcompare");
-        ecompare.setAttribute("data-pixelcompare", "");
-        ecompare.appendChild(img_now);
-        ecompare.appendChild(img_ref);
-        diff.appendChild(ecompare);
-        window.setup_pixelcompare();
-
-        const delta = document.getElementById("delta");
-        delta.style.width = img_now.naturalWidth + "px";
-        delta.style.height = img_now.naturalHeight + "px";
-        delta.appendChild(img_delta);
-    });
+
+    let loaded = { };
+
+    function load_event(tag) {
+        loaded[tag] = true;
+        if (loaded.now && loaded.ref && loaded.delta) {
+            const diff = document.getElementById("diff");
+            const w = Math.max(img_now.naturalWidth, img_ref.naturalWidth);
+            const h = Math.max(img_now.naturalHeight, img_ref.naturalHeight);
+            diff.style.width = w + "px";
+            diff.style.height = h + "px";
+            const ecompare = document.createElement("div");
+            ecompare.classList.add("pixelcompare");
+            ecompare.setAttribute("data-pixelcompare", "");
+            ecompare.appendChild(img_now);
+            ecompare.appendChild(img_ref);
+            diff.appendChild(ecompare);
+            window.setup_pixelcompare();
+
+            const delta = document.getElementById("delta");
+            delta.style.width = img_delta.naturalWidth + "px";
+            delta.style.height = img_delta.naturalHeight + "px";
+            delta.appendChild(img_delta);
+        }
+    }
+
+    img_now.addEventListener("load", () => load_event("now"));
```

Comment:
Hm, I think we want to call this load_event just once? Here we call three times, with each event.

Reply:
We want to initialize the page when all images have been loaded, without making any assumptions about which image is loaded last. See `load_event` itself how it only acts when all three are done loading.